apps/dashboard/app/views/batch_connect/session_contexts/_file_navigator.html.erb

```diff
@@ -1,4 +1,48 @@
+<%
+content_for :head do
+  javascript_include_tag 'navigator/index', nonce: true
+end
+%>
 
 <%= label attrib.id, "#{field_options[:label]}" %>
 <%= text_field :batch_connect_session_context, attrib.id, class: 'form-control' %>
-<%= button_tag 'Select Path', type: 'button', class: 'btn btn-primary', style: 'margin-top: 15px;', onclick: 'alert("Onclick Works Properly!");' %>
\ No newline at end of file
+<button type="button" class="btn btn-primary" data-toggle="modal" data-target="#modelNavigator" id='select-nav-path'>
+  Select Path
+</button>
+<span id='navigate_home_dir' style="display:none;"><%= Dir.home %></span>
+<span id='navigate_url' style="display:none;"><%= ENV['RAILS_RELATIVE_URL_ROOT'] %></span>
```

Comment:
Same comment with inline styles. You can use the bootstrap `class="d-none"`.

Also the `navigate_url` should probably be `files_path` instead of using the environment variable. I feel like using the environment variable is fragile (yes I know we use it elsewhere, we likely should change it there too).

Reply:
Thanks for the d-none. I didn't know about that one. I'll use it. I'll use the files_path as well.

Thanks!